cocos/audio/AudioEngine.cpp

```diff
@@ -210,11 +210,34 @@ int AudioEngine::play2d(const std::string& filePath, bool loop, float volume, co
             profileHelper = &_audioPathProfileHelperMap[profile->name];
             profileHelper->profile = *profile;
         }
+
+//        if (_audioIDInfoMap.size() >= _maxInstances) {
+//            log("Fail to play %s cause by limited max instance of AudioEngine",filePath.c_str());
+//            break;
+//        }
         
         if (_audioIDInfoMap.size() >= _maxInstances) {
-            log("Fail to play %s cause by limited max instance of AudioEngine",filePath.c_str());
-            break;
+            double oldestTimestamp = std::numeric_limits<double>::max();
+            int oldestId = INVALID_AUDIO_ID;
+            std::string ofilePath;
+            for(auto it = _audioIDInfoMap.begin(); it != _audioIDInfoMap.end(); ++it){
+                if(it->second.timestamp < oldestTimestamp && !it->second.loop){
```

Comment:
Please adds space after `for`, `if`, `)`, you could read https://github.com/cocos2d/cocos2d-x/blob/v3/docs/CODING_STYLE.md